File: app/src/main/java/com/softdev/smartraysam/carfeeds/Core/Presenter.java
package com.softdev.smartraysam.carfeeds.Core;

import android.content.Context;

import com.softdev.smartraysam.carfeeds.model.carModel;

import java.util.List;





/**
 * Created by Smart Raysam on 15-09-2018.
 */

public class Presenter implements GetDataContract.Presenter, GetDataContract.onGetDataListener {
    private GetDataContract.View mGetDataView;
    private Intractor mIntractor;
    public Presenter(GetDataContract.View mGetDataView){
        this.mGetDataView = mGetDataView;
        mIntractor = new Intractor(this);
    }
    @Override
    public void getDataFromURL(Context context, String url, String msg) {
        mIntractor.initRetrofitCall(context,url);
        mGetDataView.onStart(msg);
    }

    @Override
    public void onSuccess(String message, List<carModel> allData) {
        mGetDataView.onGetDataSuccess(message, allData);
    }

    @Override
    public void onFailure(String message) {
        mGetDataView.onGetDataFailure(message);
    }
}

File: app/src/main/java/com/softdev/smartraysam/carfeeds/Presenter/Presenter.kt
package com.softdev.smartraysam.carfeeds.Presenter

import android.content.Context
import com.softdev.smartraysam.carfeeds.contract.GetDataContract
import com.softdev.smartraysam.carfeeds.interactor.Intractor

import com.softdev.smartraysam.carfeeds.model.carModel

/**
 * Created by Smart Raysam on 15-09-2018.
 */

class Presenter(private val mGetDataView: GetDataContract.View) : GetDataContract.Presenter, GetDataContract.onGetDataListener {
    private val mIntractor: Intractor

    init {
        mIntractor = Intractor(this)
    }

    override fun getDataFromURL(context: Context, url: String, msg: String) {
        mIntractor.initRetrofitCall(context, url)
        mGetDataView.onStart(msg)
    }

    override fun onSuccess(message: String, allData: List<carModel>) {
        mGetDataView.onGetDataSuccess(message, allData)

    }

    override fun onFailure(message: String) {
        mGetDataView.onGetDataFailure(message)
    }
}
